rclpy/rclpy/executors.py

```diff
@@ -222,11 +227,15 @@ def _take_client(self, client):
 
     async def _execute_client(self, client, seq_and_response):
         sequence, response = seq_and_response
-        if sequence is not None and sequence == client.sequence_number:
-            # clients spawn their own thread to wait for a response in the
-            # wait_for_future function. Users can either use this mechanism or monitor
-            # the content of client.response to check if a response has been received
-            client.response = response
+        if sequence is not None:
+            try:
+                future = client._pending_requests[sequence]
+            except IndexError:
```

Comment:
This looks wrong. `client._pending_requests` is a `dict` so if `sequence` isn't a valid key a `KeyError` would be raised.

Reply:
Agreed. It looks like there's no test coverage on canceled requests.